File: Source/Rio.Web/src/app/shared/models/parcel/parcel-ledger-dto.ts
import { StringLiteralLike } from "typescript";
import { TransactionTypeDto } from "../transaction-type-dto";
import { WaterTypeDto } from "../water-type-dto";

export class ParcelLedgerDto {
    ParcelID: number;
    ParcelNumber: string;
    TransactionDate: Date;
    EffectiveDate: Date;
    TransactionType: TransactionTypeDto;
    TransactionAmount: number;
    TransactionDescription: string;
    ParcelLedgerID: number;
    WaterYear: number;
    WaterMonth: number;
    WaterType: WaterTypeDto;
    UserComment: string;

    constructor(obj?: any) {
        Object.assign(this, obj);
    }
}

File: Source/Rio.Models/DataTransferObjects/ParcelAllocation/ParcelLedgerDto.cs
using System;

namespace Rio.Models.DataTransferObjects.ParcelAllocation
{
    public class ParcelLedgerDto
    {
        public int ParcelID { get; set; }
        public string? ParcelNumber { get; set; }
        public DateTime TransactionDate { get; set; }
        public DateTime EffectiveDate { get; set; }
        public TransactionTypeDto TransactionType { get; set; }
        public decimal TransactionAmount { get; set; }
        public string TransactionDescription { get; set; }
        public int ParcelLedgerID { get; set; }
        public int WaterYear => EffectiveDate.Year;
        public int WaterMonth => EffectiveDate.Month;
        public WaterTypeDto? WaterType { get; set; }
        public string? UserComment { get; set; }
    }
}